src/main/java/ro/msg/learning/shop/service/OrderService.java

```diff
@@ -0,0 +1,100 @@
+package ro.msg.learning.shop.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.dto.StockDto;
+import ro.msg.learning.shop.exception.NoStocksAvailableException;
+import ro.msg.learning.shop.model.Address;
+import ro.msg.learning.shop.model.Order;
+import ro.msg.learning.shop.model.Stock;
+import ro.msg.learning.shop.repository.OrderRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+
+import java.util.*;
+
+@Service
+@AllArgsConstructor
+public class OrderService {
+
+    private final OrderRepository orderRepository;
+    private final ProductRepository productRepository;
+    private final StockRepository stockRepository;
+
+
+    public Order createOrder(OrderDto orderDto) {
```

Comment:
Here you either use Order as a parameter, or you return an OrderDto. But it is not ok for the method to receive an OrderDto and return Order. Usually, services work with entities and the controller is the one who does the mapping to Dto. Or there is another option in which the service receives Dto and returns again a Dto and the mapping to the domain is done in the methods inside the service. But there is never a mix between the two options